pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -389,18 +433,60 @@ func (a *Assignment) findFlavorForResourceGroup(
 			}
 		}
 
-		if representativeMode > bestAssignmentMode {
-			bestAssignment = assignments
-			bestAssignmentMode = representativeMode
-			if bestAssignmentMode == Fit {
-				// All the resources fit in the cohort, no need to check more flavors.
-				return bestAssignment, nil
+		if features.Enabled(features.FlavorFungibility) {
+			if !shouldTryNextFlavor(representativeMode, cq.FlavorFungibility, needsBorrowing) {
+				bestAssignment = assignments
+				bestAssignmentMode = representativeMode
+				break
+			}
+			if representativeMode > bestAssignmentMode {
+				bestAssignment = assignments
+				bestAssignmentMode = representativeMode
+			}
+		} else {
+			if representativeMode > bestAssignmentMode {
+				bestAssignment = assignments
+				bestAssignmentMode = representativeMode
+				if bestAssignmentMode == Fit {
+					// All the resources fit in the cohort, no need to check more flavors.
+					return bestAssignment, nil
+				}
 			}
 		}
 	}
+
+	for _, assignment := range bestAssignment {
+		if flavorIdx == len(rg.Flavors)-1 {
+			// we have reach the last flavor, try from the first flavor next time
+			assignment.FlavorIdx = -1
+		} else {
+			assignment.FlavorIdx = flavorIdx
+		}
+	}
+	if bestAssignmentMode == Fit {
+		return bestAssignment, nil
+	}
```

Comment:
Are these needed if the `FlavorFungibility` featuregate is disabled?

Reply:
done